runtime/tests/checker/genericfunction_test.go

```diff
@@ -939,3 +939,167 @@ func TestCheckUnparameterizedTypeInstantiationE(t *testing.T) {
 
 	assert.IsType(t, &sema.UnparameterizedTypeInstantiationError{}, errs[0])
 }
+
+func TestCheckGenericFunctionDeclaration(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("global, struct", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+          fun head<T: AnyStruct>(_ items: [T]): T? {
+              if items.length < 1 {
+                  return nil
+              }
+              return items[0]
```

Comment:
I think we should have some more tests for these; in particular I'd like to see tests for when a generically-typed value "escapes" the scope of its function into an external/global variable. E.g. if I have a function like this that instead stores `items[0]` in a global array of type `[AnyStruct] vs one of type `[String]`, and how these subtyping checks interact with the type parameter bound.